homework/list.js

```diff
@@ -1,79 +1,76 @@
 'use strict';
 
-(function() {
+function ListManager() {
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper');
+}
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
-
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
-
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
+ListManager.prototype = {
+    start: function () {
+        this.fetchList().then(function (data) {
+            this.updateList(data);
+        }.bind(this)).then(this.drawList.bind(this)).then(this.preloadFirstNote.bind(this));
+        window.addEventListener('click', function (event) {
+            this.onNoteOpen(event);
+        }.bind(this));
+    },
+    onNoteOpen: function (event) {
```

Comment:
Well it's a convention to prefix with on to handle the corresponding event, rather than to fire it. So, you should rename this function as onClick and to dispatch the note-open.